src/ui/atoms/button-icon/index.tsx

```diff
@@ -2,99 +2,115 @@ import * as React from 'react';
- * --woly-line-height
- *
- * --woly-background
- * --woly-border
- *
- * --woly-background-hover
- * --woly-border-hover
- *
- * --woly-background-focus
- * --woly-border-focus
- *
- * --woly-background-disabled
- * --woly-border-disabled
- */
-
 interface Props extends React.ButtonHTMLAttributes<HTMLButtonElement> {
   className?: string;
   icon: React.ReactNode;
   onClick: React.MouseEventHandler<HTMLButtonElement>;
+  filled?: boolean;
 }
 
 const ButtonIconBase: React.FC<Props & Variant> = ({
   icon,
   onClick,
+  filled = false,
   variant = 'default',
   ...p
-}) => {
-  const onClickHandler = React.useCallback(
-    (event: React.MouseEvent<HTMLButtonElement>) => {
-      event.preventDefault();
-      onClick(event);
-    },
-    [onClick],
-  );
-
-  return (
-    <button data-variant={variant} onClick={onClickHandler} type="button" {...p}>
-      <span>{icon}</span>
-    </button>
-  );
-};
+}) => (
+  <button data-filled={filled} data-variant={variant} onClick={onClick} type="button" {...p}>
+    <span data-icon>{icon}</span>
+  </button>
+);
 
 export const ButtonIcon = styled(ButtonIconBase)`
-  --woly-vertical: calc(1px * var(--woly-component-level) * var(--woly-main-level));
-  --woly-horizontal: calc(
-    var(--woly-const-m) + (1px * var(--woly-main-level)) + var(--woly-vertical)
-  );
-  --woly-gap: calc(
-    (1px * var(--woly-main-level)) + (1px * var(--woly-main-level) * var(--woly-component-level))
-  );
-
-  display: flex;
-  flex-wrap: nowrap;
-  padding: var(--woly-vertical, 6.4px) var(--woly-horizontal, 6.4px);
-
-  background-color: var(--woly-background, transparent);
-  border-color: var(--woly-border, #000000);
-  border-style: solid;
-  border-width: var(--woly-border-width, 0);
+  --local-shape-color: var(--woly-canvas-default);
+  --local-icon-size: var(--woly-line-height);
+  --local-icon-color: var(--woly-canvas-text-default);
+  --local-padding: calc(1px * var(--woly-component-level) * var(--woly-main-level));
 
+  border: var(--woly-border) solid var(--local-shape-color);
   border-radius: var(--woly-rounding, 4px);
   cursor: pointer;
+  padding: var(--local-padding);
+  background: var(--local-shape-color);
+  outline: none;
 
-  span {
+  [data-icon] {
     display: flex;
     align-items: center;
     justify-content: center;
-    width: var(--woly-line-height, 24px);
-    height: var(--woly-line-height, 24px);
+    width: var(--local-icon-size);
+    height: var(--local-icon-size);
+
+    svg > path {
+      fill: var(--local-icon-color);
+    }
+    svg > g {
+      stroke: var(--local-icon-color);
+    }
   }
 
-  &:hover {
-    background-color: var(--woly-background-hover, #000000);
-    border-color: var(--woly-border-hover, #000000);
-    outline: none;
+  &[data-filled='true'] {
+    --local-shape-color: var(--woly-shape-default);
+    svg > path {
+      --local-icon-color: var(--woly-shape-text-default);
+    }
+    svg > g {
+      --local-icon-color: var(--woly-shape-text-default);
+    }
```

Comment:
```suggestion
    --local-icon-color: var(--woly-shape-text-default);
```

А зачем селекторы здесь?

Переменные же провалятся прям внутрь

Reply:
Да, точно 👍